### ISSUE TITLE
List of Songs to be added to the queue

### ISSUE THREAD

**Issue by daxcurson** [daxcurson/ampflower#4]
When I select an album, an artist or a playlist, I want to see the songs that it has, so that I can add them to the queue of songs.

**Comment by daxcurson** [comment]
This seems to be working, but now has a bug: the actual list is filled up two times, and the second time, the values of album and artist are null. I think this is because the first time, the data is pulled from the network, which does include the names of the artist and album with the song retrieved, and the second time is because the data is pulled from the database. Now, what I tried was remove the notifyDataSetChanged(), which prompts the RecyclerView to update its contents. But then, although the method to fill the list is called only once, now the items aren't displayed at all, which puzzles me... Why, oh, why????

**Comment by daxcurson** [comment]
I solved the problem of the recycler containing the same songs 2 times. The problem was that the network retrieval was subscribed on a io thread but the observer thread had not been configured. I had actually commented-out the .observeOn call when the application gave an error saying that the database couldn't be read in the main thread, but something else must have been the cause of that. Now I removed the comment marking on the .observeOn causing the main thread to observe the scheduler io, and the items were added just once... the .observe() call in ChooseSongsFragment was actually called 2 times before, and now it is called just once. Maybe, because of not assigning the .observeOn, the observers were in total 2, one that was by default on the mainThread, and another one in the io thread???? Well, I don't know, now it's working.
